nni/experiment/experiment.py

```diff
@@ -128,12 +119,27 @@ def start(self, port: int = 8080, debug: bool = False, run_mode: RunMode = RunMo
         ips = [f'http://{ip}:{port}' for ip in ips if ip]
         msg = 'Web portal URLs: ' + colorama.Fore.CYAN + ' '.join(ips) + colorama.Style.RESET_ALL
         _logger.info(msg)
+        return config
```

Comment:
Is the return value ever used?
If no I suggest to remove it because I will not expect a function named "start" to return a config.
We can add it back any time when we really need it.

Reply:
this is not `start`, it is `_start_impl`. they are different. `start` may include more logic than `_start_impl`, the canonicalize is called in `_start_impl`, it would be better for it to return the canonicalized config for the following usage. It is used in RetiariiExperiment